libfwupdplugin/fu-quirks.c

```diff
@@ -174,9 +174,9 @@ fu_quirks_add_quirks_for_path (FuQuirks *self, XbBuilder *builder,
 		g_autoptr(XbBuilderSource) source = xb_builder_source_new ();
 
 		/* load from keyfile */
-		xb_builder_source_add_adapter (source, "text/plain,.quirk",
-					       fu_quirks_convert_quirk_to_xml_cb,
-					       NULL, NULL);
+		xb_builder_source_add_simple_adapter (source, "text/plain,.quirk",
```

Comment:
To avoid forcing a hard dependency on the updated libxmlb, can you please add a `#define` version guard?  If on the newer libxmlb use this, and if not then use the older method.